src/tool/GenerateAction.cpp

```diff
@@ -13,15 +13,108 @@
 #include "lib/Lib/AbsoluteCompilationDatabase.hpp"
 #include "lib/Lib/ConfigImpl.hpp"
 #include "lib/Lib/CorpusImpl.hpp"
+#include "llvm/Support/Program.h"
 #include <mrdocs/Generators.hpp>
 #include <mrdocs/Support/Error.hpp>
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
+
 #include <cstdlib>
+#include <unordered_map>
 
 namespace clang {
 namespace mrdocs {
 
+std::optional<std::string> 
+getCompilerInfo(llvm::StringRef compiler) 
+{
+    llvm::SmallString<128> outputPath;
+    if (auto EC = llvm::sys::fs::createTemporaryFile("compiler-info", "txt", outputPath)) 
+    {
+        return std::nullopt;
+    }
+
+    std::optional<llvm::StringRef> redirects[] = {llvm::StringRef(), llvm::StringRef(), outputPath.str()};
+    std::vector<llvm::StringRef> args = {compiler, "-v", "-E", "-x", "c++", "-"};
+
+    llvm::ErrorOr<std::string> compilerPath = llvm::sys::findProgramByName(compiler);
+    if ( ! compilerPath) 
+    {
+        return std::nullopt;
+    }
+
+    int result = llvm::sys::ExecuteAndWait(*compilerPath, args, std::nullopt, redirects);
```

Comment:
`compiler` is set as the first arg so I have the impression that the compiler path might be passed twice here.

`std::nullopt` makes it inherit the parent environment. Is that the same behavior as CMake (or the behavior we want)? Otherwise, we need to replace it with an empty `ArrayRef<StringRef>`.

Does LLVM provide a function where we can we capture the result directly in a string?

I'm also kind of impressed `std::optional<llvm::StringRef>` is implicitly convertible to `ArrayRef<std::optional<StringRef>>`.

Reply:
> compiler is set as the first arg so I have the impression that the compiler path might be passed twice here.

You're right to double-check the arguments for llvm::sys::ExecuteAndWait. According to the [LLVM documentation](https://llvm.org/doxygen/namespacellvm_1_1sys.html#abb13fbc7f52c659966b05092b19e6e19), the first element in the Args array should indeed be the name of the program.

Reply:
> std::nullopt makes it inherit the parent environment. Is that the same behavior as CMake (or the behavior we want)? Otherwise, we need to replace it with an empty ArrayRef<StringRef>.

I've updated the code to use an empty llvm::ArrayRef<llvm::StringRef> instead, ensuring that the child process does not inherit the environment variables from the parent process.

Reply:
> Does LLVM provide a function where we can we capture the result directly in a string?

After extensive testing with various approaches, I've found that the current implementation is the only one that consistently works as intended.

Reply:
> I'm also kind of impressed std::optional<llvm::StringRef> is implicitly convertible to ArrayRef<std::optional<StringRef>>.

Indeed, the `redirects` variable is an array of three elements, each of type `std::optional<StringRef>`